### PR TITLE
Exception handling

### PR DESCRIPTION
**Pull request by bkuhl**
* Adds exception handling to the webhook handler and friendly exception messages (where possible) to assist in more easily recognizing error messages.  Instead of having to parse the response to determine the problem, applications will just catch a `TooManyWebhooks` exception
 * Adds necessary middleware to the webhook creation route to ensure oauth token is registered correctly with the api client
 * If a person has not yet configured `webhook-created-handler` correctly it'll fail before creating the webhook, preventing a webhook from actually being created unless there's a handler to go alongside it

### CODE REVIEW

**Comment by bkuhl** [comment]
Several breaking changes are introduced in this PR, but we're not yet at v1, so I think that's ok.

File: src/Discord/ErrorFactory.php [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+namespace LaravelRestcord\Discord;
+
+use LaravelRestcord\Discord\Webhooks\TooManyWebhooks;
+use LaravelRestcord\Exception;
+
+class ErrorFactory
+{
+    public function make(int $code, string $message) : ?Exception
+    {
+        if ($code == 30007) {
```
Comment (alexknutson) [inline src/Discord/ErrorFactory.php]: hot. Should this be in a constant? 🎱

File: tests/Http/WebhookCallbackTest.php [hunk 2]
```diff
@@ -96,7 +102,88 @@ public function createsWebhook()
...
+        ]));
+        $response = Mockery::mock(Response::class);
+        $response->shouldReceive('getBody')->andReturn($stream);
+
+        $exception->shouldReceive('getResponse')->andReturn($response);
+        $this->client->shouldReceive('post')->andThrow($exception);
+
+        $expectedResponse = uniqid();
+        $handlesWebhookCreated = Mockery::mock(HandlesDiscordWebhooksBeingCreated::class);
+        // for some reason this mock value isn't obeyed and it's pulling the actual class
```
Comment (alexknutson) [inline tests/Http/WebhookCallbackTest.php]: So do we need to keep it?

File: src/Discord/ErrorFactory.php (hunk 1 above)
Comment (bkuhl) [inline src/Discord/ErrorFactory.php]: Great idea!  It's been updated

File: tests/Http/WebhookCallbackTest.php (hunk 2 above)
Comment (bkuhl) [inline tests/Http/WebhookCallbackTest.php]: I'd like to keep it because I still think it needs to be addressed.

**Pull request merged**